contracts/sol6/bridgeReserve/balancer/KyberBalancerReserve.sol

```diff
@@ -0,0 +1,253 @@
+        } else {
+            destToken.safeTransfer(destAddress, expectedDestAmount);
+        }
+
+        emit TradeExecute(
+            msg.sender,
+            srcToken,
+            srcAmount,
+            destToken,
+            expectedDestAmount,
+            destAddress
+        );
+        return true;
+    }
+
+    function setKyberNetwork(address _kyberNetwork) external onlyAdmin {
+        require(_kyberNetwork != address(0));
+        if (kyberNetwork != _kyberNetwork) {
+            kyberNetwork = _kyberNetwork;
+            emit KyberNetworkSet(kyberNetwork);
+        }
+    }
+
+    function setBalancerExchangeProxy(
+        IBalancerExchangeProxy _newExchangeProxy
+    )
+        external onlyAdmin
+    {
+        require(_newExchangeProxy != IBalancerExchangeProxy(0), "new proxy is 0");
+        if (exchangeProxy != _newExchangeProxy) {
+            exchangeProxy = _newExchangeProxy;
+            emit BalancerExchangeProxySet(_newExchangeProxy);
+        }
+    }
+
+    function listTokens(
+        IERC20[] calldata tokens,
+        uint256[] calldata nPools
+    )
+        external onlyOperator
+    {
+        require(tokens.length == nPools.length, "lengths must be the same");
+        for(uint256 i = 0; i < tokens.length; i++) {
+            listToken(tokens[i], nPools[i]);
+        }
+    }
+
+    function delistTokens(IERC20[] calldata tokens) external onlyOperator {
+        for(uint256 i = 0; i < tokens.length; i++) {
+            delistToken(tokens[i]);
+        }
+    }
+
+    function updateNumberPoolsForTokens(
+        IERC20[] calldata tokens, 
+        uint256[] calldata nPools
+    )
+        external onlyOperator
+    {
+        require(tokens.length == nPools.length, "lengths must be the same");
+        for(uint256 i = 0; i < tokens.length; i++) {
+            require(numberPools[tokens[i]] > 0, "token is not listed yet");
+            require(nPools[i] > 0, "number pools must be positive");
+            numberPools[tokens[i]] = nPools[i];
+        }
+        emit NumberPoolsForTokensUpdated(tokens, nPools);
+    }
+
+    function enableTrade() external onlyAdmin {
+        tradeEnabled = true;
+        emit TradeEnabled(true);
+    }
+
+    function disableTrade() external onlyAlerter {
+        tradeEnabled = false;
+        emit TradeEnabled(false);
+    }
+
+    /**
+     *   @dev called by kybernetwork to get settlement rate
+     */
+    function getConversionRate(
+        IERC20 src,
+        IERC20 dest,
+        uint256 srcQty,
+        uint256 /* blockNumber */
+    ) external override view returns (uint256 rate) {
+        if (!tradeEnabled) return 0;
+        if (srcQty == 0) return 0;
+
+        uint256 destQty;
+        if (src == ETH_TOKEN_ADDRESS) {
+            uint256 nPools = numberPools[dest];
+            if (nPools == 0) return 0; // token is not listed
+            (, destQty) = exchangeProxy.viewSplitExactIn(
+                weth,
```

Comment:
here should use weth while in trade can use eth (0xeeee....?)
not clear why this diff.

Reply:
yes this is how it works, the viewSplitExactIn func only accepts weth address, but the swap function supports eth addres